src/main/java/project/service/web/SmarttenderService.java

```diff
@@ -0,0 +1,118 @@
+package project.service.web;
+
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
+import project.model.dto.smarttender.SmarttenderDTO;
+import project.model.response.Initiator;
+import project.model.response.Site;
+import project.model.response.Tender;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SmarttenderService implements IWebService {
+    private List<project.model.dto.smarttender.Tender> tenderList;
+    private List<project.model.dto.smarttender.Tender> onlyNewTenderList;
+
+    private final String BASE_URL = "https://smarttender.biz/CommercialTrades/GetTenders/";
```

Comment:
а чи не треба ось такі URL виносити в Additional.properties або в System.getEnv()?

Reply:
Можливо.
Тут логіка була, щоб усе, що відноситься до роботи з одним сайтом, залишалось в одному класі.

Reply:
@Dmytro-Vusyk could you please help us here?

Reply:
It can be implemented as it is. But in practice, all the external URLs must be stored as a property. Because you may have different URLs depending from env. Production, qa etc

Reply:
by the way, BASE_URL is your host URL https://smarttender.biz - this is the base.